src/mon/OSDMonitor.cc

```diff
@@ -7111,6 +7111,26 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
     ss << osdmap.get_crush_version() + 1;
     goto update;
 
+  } else if (prefix == "osd crush set-all-straw-buckets-to-straw2") {
+    CrushWrapper newcrush;
+    _get_pending_crush(newcrush);
+    for (int b = 0; b < newcrush.get_max_buckets(); ++b) {
+      int bid = -1 - b;
+      if (newcrush.bucket_exists(bid) &&
+	  newcrush.get_bucket_alg(bid)) {
```

Comment:
shall we check if the returned alg is `CRUSH_BUCKET_STRAW` ? and if it's not or is already `CRUSH_BUCKET_STRAW2`, we might want to ignore it. and only `wait_for_finished_proposal()` if we changed the alg of any bucket.

Reply:
fixed the check.

Making a no-op path is a lot of work because the get_pending_crush thing may return an unstable crush map, and you still have to wait if it isn't the committed one.  That means also having a flag for whether it is the stable or pending one or not (which we don't have currently) or repeating all the checks on osdmap.crush.